ocp-metadata/ocp-metadata.go

```diff
@@ -57,16 +58,18 @@ func (meta *Metadata) GetClusterMetadata() (ClusterMetadata, error) {
 	if err != nil {
 		return metadata, nil
 	}
-	metadata.ClusterName, metadata.Platform, metadata.Region = infra.Status.InfrastructureName, infra.Status.Platform, infra.Status.PlatformStatus.Aws.Region
-	metadata.ClusterType = "self-managed"
-	for _, v := range infra.Status.PlatformStatus.Aws.ResourceTags {
-		if v.Key == "red-hat-clustertype" {
-			metadata.ClusterType = v.Value
+	if infra != nil {
+		metadata.ClusterName, metadata.Platform, metadata.Region = infra.Status.InfrastructureName, infra.Status.Platform, infra.Status.PlatformStatus.Aws.Region
+		metadata.ClusterType = "self-managed"
+		for _, v := range infra.Status.PlatformStatus.Aws.ResourceTags {
+			if v.Key == "red-hat-clustertype" {
+				metadata.ClusterType = v.Value
+			}
+		}
+		metadata.SDNType, err = meta.getSDNInfo()
```

Comment:
Do we need `SDNType` as well to be populated only when there are infra nodes present?

Reply:
nope, that's an errata, fixing

Reply:
infra does not refer to infra nodes but to the "infrastructure" object, which is only present in OpenShift clusters, same as `config.openshift.io/v1/networks`